internal/app/streamer.go

```diff
@@ -25,78 +23,51 @@ type Replicator interface {
 type BasinStreamer struct {
 	namespace  string
 	replicator Replicator
-	privateKey *ecdsa.PrivateKey
-	provider   BasinProvider
+	dbMngr     *DBManager
 }
 
 // NewBasinStreamer creates new streamer.
-func NewBasinStreamer(ns string, r Replicator, bp BasinProvider, pk *ecdsa.PrivateKey) *BasinStreamer {
+func NewBasinStreamer(ns string, r Replicator, dbm *DBManager) *BasinStreamer {
 	return &BasinStreamer{
 		namespace:  ns,
 		replicator: r,
-		provider:   bp,
-		privateKey: pk,
+		dbMngr:     dbm,
 	}
 }
 
 // Run runs the BasinStreamer logic.
 func (b *BasinStreamer) Run(ctx context.Context) error {
-	txs, table, err := b.replicator.StartReplication(ctx)
+	// Open a local DB for replaying txs
+	db, err := b.dbMngr.NewDB()
+	if err != nil {
+		return err
+	}
+	b.dbMngr.db = db
+
+	// Create sink table in local DB
+	if err := b.dbMngr.Setup(ctx); err != nil {
+		return fmt.Errorf("cannot setup db: %s", err)
+	}
+	defer func() {
+		_ = b.dbMngr.Close()
+	}()
+
+	// Start replication
+	txs, _, err := b.replicator.StartReplication(ctx)
 	if err != nil {
 		return fmt.Errorf("start replication: %s", err)
 	}
 
 	for tx := range txs {
 		slog.Info("new transaction received")
-
-		capnpTx, err := basincapnp.FromPgReplTx(tx)
-		if err != nil {
-			return fmt.Errorf("to capnproto: %s", err)
-		}
-
-		signature, err := b.sign(capnpTx)
-		if err != nil {
-			return fmt.Errorf("sign: %s", err)
+		if err := b.dbMngr.Replay(ctx, tx); err != nil {
+			return fmt.Errorf("replay: %s", err)
 		}
-
-		if err := b.push(ctx, table, capnpTx, signature); err != nil {
-			return fmt.Errorf("push: %s", err)
-		}
-
 		if err := b.replicator.Commit(ctx, tx.CommitLSN); err != nil {
 			return fmt.Errorf("commit: %s", err)
 		}
-
 		slog.Info("transaction acked")
 	}
 
 	return nil
 }
-
-func (b *BasinStreamer) sign(tx basincapnp.Tx) ([]byte, error) {
```

Comment:
we wouldn't need `sign` and `push` because we can re use the file `uploader`'s functions